packages/calcite-components/src/components/list/list.tsx

```diff
@@ -937,20 +954,44 @@ export class List extends LitElement implements InteractiveComponent, SortableCo
     const toEl = moveTo.element as List["el"];
     const fromElItems = Array.from(fromEl.children).filter(isListItem);
     const oldIndex = fromElItems.indexOf(dragEl);
+    const newIndex = 0;
 
     if (!fromEl) {
       return;
     }
 
+    if (
```

Comment:
Since this is shared, let’s create a utility for both `list` and `block-list` to use, and do the same for their tests.

Reply:
Can we do this as part of a new refactor issue?

Reply:
Sounds good, but let’s make it a priority to unify these paths as soon as possible.

Reply:
created https://github.com/Esri/calcite-design-system/issues/11849